test/Dialect/FIRRTL/annotations.mlir

```diff
@@ -1495,3 +1495,35 @@ firrtl.circuit "TraceNameAnnotation" attributes {rawAnnotations = [
     %w = firrtl.wire : !firrtl.uint<1>
   }
 }
+
+// -----
+
+// Test that the valid types are connected, when the source has un-inferred width but sink has width.
+firrtl.circuit "Top"  attributes {rawAnnotations = [
+	{
+		class = "sifive.enterprise.grandcentral.DataTapsAnnotation",
+		keys = [
+						{
+						class = "sifive.enterprise.grandcentral.ReferenceDataTapKey",
+						sink = "~Top|Top>tap",
+						source = "~Top|Foo>sum"
+			}]}]} {
+  firrtl.module private @Foo(in %in: !firrtl.uint<8>, out %out: !firrtl.uint<8>) {
+    %c1_ui = firrtl.constant 1 : !firrtl.uint
+    %0 = firrtl.add %in, %c1_ui : (!firrtl.uint<8>, !firrtl.uint) -> !firrtl.uint
+    %1 = firrtl.tail %0, 1 : (!firrtl.uint) -> !firrtl.uint
+    %sum = firrtl.node interesting_name %1  : !firrtl.uint
+    firrtl.connect %out, %sum : !firrtl.uint<8>, !firrtl.uint
+  }
+  // CHECK-LABEL: firrtl.module @Top
+  firrtl.module @Top(in %in: !firrtl.uint<8>, out %out: !firrtl.uint<8>) {
+    %foo_in, %foo_out = firrtl.instance foo interesting_name  @Foo(in in: !firrtl.uint<8>, out out: !firrtl.uint<8>)
+    firrtl.strictconnect %foo_in, %in : !firrtl.uint<8>
+    %tap = firrtl.wire interesting_name  : !firrtl.uint<8>
+    %0 = firrtl.add %foo_out, %tap : (!firrtl.uint<8>, !firrtl.uint<8>) -> !firrtl.uint<9>
+    %1 = firrtl.tail %0, 1 : (!firrtl.uint<9>) -> !firrtl.uint<8>
+    firrtl.strictconnect %out, %1 : !firrtl.uint<8>
+    // CHECK:   %0 = firrtl.ref.resolve %foo__gen_tap : !firrtl.ref<uint>
+    // CHECK:   firrtl.connect %tap, %0 : !firrtl.uint<8>, !firrtl.uint
+  }
+}
```

Comment:
This test is pretty big. Can it be slimmed down to remove all the unnecessary things, e.g., `firrtl.add`/`firrtl.tail`?

Reply:
Thanks @seldridge , simplified the test.